### PR TITLE
Updating readme with detailed instructions

### PR DESCRIPTION
**Pull request by spurohitsynopsys**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -130,81 +161,101 @@ jobs:
...
 
-On pushes, a full "intelligent" Black Duck scan will be run. On pull requests, a "rapid" ephemeral scan will be run.
-A future release of the action will provide code review feedback for newly introduced findings to the pull request.
+In the default Black Duck Hub permission model, projects and project versions are created on the fly as needed.
 
-We recommend configuring sensitive data like access tokens and even URLs, using GitHub secrets.
+The action will download the Bridge and Detect CLIs, run a SCA scan, and optionally break the build.
 
+On pushes, a full **Intelligent** Black Duck scan will be run. On pull requests, a **Rapid** ephemeral scan will be run.
+A future release of the action will provide code review feedback for newly introduced findings to the pull request.
```
Comment (sig-tithi) [inline README.md:177]: for BD and coverity we already have pr comment so  A future release of the action will provide code review feedback for newly introduced findings to the pull request.  we can remove. same for coverity

File: README.md [hunk 2]
```diff
@@ -18,10 +18,33 @@ In this Quick Start, we will provide individual examples for each Synopsys secur
...
 - Synopsys solution functionality is invoked directly by the Synopsys Bridge, and indirectly by the Synopsys Action
 
+# Prerequisites
+
+Before configuring Synopsys Action into your workflow, note the following prerequisites:
+
+**GitHub Runner Setup:**
+
+- Runners are the machines that execute jobs in a GitHub Actions workflow. To use GitHub runners in your project, GitHub Actions must be enabled for a repository/organization settings in order for required workflows to run (Repository Settings → SelectActions → General → Actions permissions)
+- GitHub runner can be Self-hosted or GitHub-hosted. For installing Self-hosted runners refer [Self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners) and for installing GitHub-hosted runners refer [GitHub-hosted Runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners)
```
Comment (kishorikumar) [inline README.md:31]: We cannot install github-hosted runner. So may be we can rephrase "installing Github-hosted runners" to "using Github hosted runner"

File: README.md (hunk 1 above)
Comment (spurohitsynopsys) [reply README.md:176]: updated

File: README.md [hunk 3]
```diff
@@ -18,10 +18,33 @@ In this Quick Start, we will provide individual examples for each Synopsys secur
...
+**Configure GitHub Token:** 
+
+- `github_token` is required as input when running Black Duck Fix PR, Black Duck/Coverity PR Comment. There are 2 different types of tokens that can be passed to `github_token`:
+
+  1. Token can be github specified `secrets.GITHUB_TOKEN` with required workflow read & write permissions(GitHub → Project → Settings → Actions → General → Workflow Permissions). It will be created by GitHub at start of each workflow run.
+  2. If you need a token that requires permissions that aren't available in the `secrets.GITHUB_TOKEN`, you can create a Personal Access Token(PAT) with required scopes (Select Profile Photo → Settings → Developer Settings → Personal access tokens). For more information, refer [ Granting Additional Permissions ](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#granting-additional-permissions) <br/> PAT must have **repo** scope to perform Black Duck Fix PR, Black Duck/Coverity PR Comment.
+
+**Create workflow:**
+
+- Create a new workflow (GitHub → Project → Actions → New Workflow → Setup a workflow yourself) and configure required fields.
```
Comment (kishorikumar) [inline README.md:45]: Are we mentioning anywhere that we can select synopsys-action from marketplace on workflow configuration page and see synopsys-action configuration along with it's parameter for help? 
If no, shall we mention that as well in our readme.

File: README.md [hunk 4]
```diff
@@ -214,28 +265,11 @@ observe fewer pull requests to be created.**
...
-
-# Future Enhancements
+| `include_diagnostics`      | Synopsys Bridge diagnostics files will be available to download when it is set to `true`. Additionally **diagnostics_retention_days** can be passed as integer value between 1 to 90 to retain the files (Be default file be available for 90 days).               |
 
-- Provide comments on Pull Requests about code quality issues.
-- Prevent a merge if security issues are found during the pull request. Create a GitHub status check and report the policy as failed if new security issues are found.
-- Create GitHub Issues to track issues found by a full analysis. This action is currently focused on providing feedback on a pull request. No action is taken if run manually or on a push. A future enhancement is to create GitHub issues to track security weaknesses found during a push.
-- Allow developers to dismiss issues from the pull request. If an issue is deemed to be a false positive, a future enhancement could allow the developer to indicate this by replying to the comment, which would in turn report the status to the Coverity Connect instance so that future runs will recognize the issue as having been triaged as such.
+**Notes:**
```
Comment (kishorikumar) [inline README.md:270]: I think we should mention caching strategy as well as part of note.

**Review by kishorikumar — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (spurohitsynopsys) [reply README.md:31]: https://docs.github.com/en/enterprise-cloud@latest/actions/hosting-your-own-runners/managing-self-hosted-runners/adding-self-hosted-runners#adding-a-self-hosted-runner-to-an-organization
GitHub is referring it as installation only

File: README.md (hunk 3 above)
Comment (spurohitsynopsys) [reply README.md:45]: update before blackduck, coverity and polaris example
we did same in extension

File: README.md (hunk 4 above)
Comment (spurohitsynopsys) [reply README.md:270]: as discussed with tithi, it will be updated as part of airgap

**Review by kishorikumar — APPROVED** [review]

**Review by sig-tithi — APPROVED** [review]

**Pull request merged**